ledger-tool/src/blockstore.rs

```diff
@@ -891,26 +891,24 @@ fn do_blockstore_process_command(ledger_path: &Path, matches: &ArgMatches<'_>) -
                 AccessType::PrimaryForMaintenance,
             );
 
-            let end_slot = match end_slot {
-                Some(end_slot) => end_slot,
-                None => {
-                    let slot_meta_iterator = blockstore.slot_meta_iterator(start_slot)?;
-                    let slots: Vec<_> = slot_meta_iterator.map(|(slot, _)| slot).collect();
-                    if slots.is_empty() {
-                        return Err(LedgerToolError::BadArgument(format!(
-                            "blockstore is empty beyond purge start slot {start_slot}"
```

Comment:
This previously got an iterator from `start_slot` onwards and would error if the collected vector was empty. This could happen in two cases:
1. The blockstore is completely empty
2. Start slot is greater than the blockstore's highest slot

1. is covered below by the new check ensuring `highest_slot().is_some()`, and 2. is covered with ensuring that `start_slot <= end_slot`